hosted/helpers/helper_charts.go

```diff
@@ -0,0 +1,139 @@
+package helpers
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/rancher-sandbox/ele-testhelpers/kubectl"
+	"github.com/rancher-sandbox/ele-testhelpers/tools"
+	"github.com/rancher/shepherd/clients/rancher/catalog"
+)
+
+// AddRancherCharts adds the repo from which rancher can be installed
+func AddRancherCharts() {
+	err := kubectl.RunHelmBinaryWithCustomErr("repo", "add", catalog.RancherChartRepo, "https://charts.rancher.io")
+	Expect(err).To(BeNil())
+	err = kubectl.RunHelmBinaryWithCustomErr("repo", "add", fmt.Sprintf("rancher-%s", RancherChannel), fmt.Sprintf("https://releases.rancher.com/server-charts/%s", RancherChannel))
+	Expect(err).To(BeNil())
+}
+
+// GetCurrentOperatorChartVersion returns the current version of a Provider chart.
+func GetCurrentOperatorChartVersion() string {
+	charts := ListOperatorChart()
+	if len(charts) == 0 {
+		return ""
+	}
+	return charts[0].DerivedVersion
+}
+
+// GetDowngradeOperatorChartVersion returns a version to downgrade to from a given chart version.
+func GetDowngradeOperatorChartVersion(currentChartVersion string) string {
+	var chartName string
+	if charts := ListOperatorChart(); len(charts) > 0 {
+		chartName = charts[0].Name
+	} else {
+		ginkgo.GinkgoLogr.Info("Could not find downgrade chart; chart is not installed")
+		return ""
+	}
+	chartVersions := ListChartVersions(chartName)
+	for _, chartVersion := range chartVersions {
+		if VersionCompare(chartVersion.DerivedVersion, currentChartVersion) == -1 {
+			return chartVersion.DerivedVersion
+		}
+	}
+	return ""
+}
+
+func DowngradeProviderChart(downgradeChartVersion string) {
+	currentChartVersion := GetCurrentOperatorChartVersion()
+	Expect(currentChartVersion).ToNot(BeEmpty())
+	UpdateOperatorChartsVersion(downgradeChartVersion)
+	Expect(VersionCompare(downgradeChartVersion, currentChartVersion)).To(Equal(-1))
+}
+
+// WaitUntilOperatorChartInstallation waits until the current operator chart version compares to the input chartVersion
+// compareTo value can be 0 if current == input; -1 if current < input; 1 if current > input; defaults to 0
+// compareTo is helpful in case either of the chart version is unknown;
+// for e.g. after a rancher upgrade, if only the old chart version is known then we can wait until the current chart version is greater than it
```

Comment:
```suggestion
// WaitUntilOperatorChartInstallation waits until the current operator chart version compares to the input chartVersion using the comparator
// comparator values can be >,<,<=,>=,==,!=
// compareTo value can be 0 if current == input; -1 if current < input; 1 if current > input; defaults to 0
// compareTo is helpful in case either of the chart version is unknown;
// for e.g. after a rancher upgrade, if only the old chart version is known then we can wait until the current chart version is greater than it
```